aws/resource_aws_route_table_test.go

```diff
@@ -555,6 +555,16 @@ func TestAccAWSRouteTable_ConditionalCidrBlock(t *testing.T) {
 	})
 }
 
+func testAccCheckAWSRouteTableNumberOfRoutes(routeTable *ec2.RouteTable, n int) resource.TestCheckFunc {
```

Comment:
Shared with https://github.com/terraform-providers/terraform-provider-aws/pull/14013.

Reply:
We should try to move away from API object `TestCheckFunc` like these in preference of state value and ImportStateVerify checks -- please see: https://github.com/hashicorp/terraform-provider-aws/issues/16435

Reply:
In general I completely agree that those kind of checkers aren't desirable, in this case I am verifying that the number of routes returned by the API is _NOT_ equal to the number of routes stored in state:

```go
testAccCheckAWSRouteTableNumberOfRoutes(&routeTable, 2),
resource.TestCheckResourceAttr(resourceName, "route.#", "1"),
```

effectively testing the validity of this code:

https://github.com/hashicorp/terraform-provider-aws/blob/346cac93aba3fd8f87b2de981c9213b74c44ad86/aws/resource_aws_route_table.go#L211-L224

Reply:
Happy to remove if you think this is overkill.